sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -469,6 +469,189 @@ def test_empty_output_box(gallery_conf):
     assert output.isspace()
 
 
+code_repr_only = (
+    'class repr_only_class():\n'
+    '    def __init__(self):\n'
+    '        pass\n\n'
+    '    def __repr__(self):\n'
+    '        return "This is the __repr__"\n\n'
+    'class_inst = repr_only_class()\n\n'
+    'class_inst'
+)
+
+code_repr_and_html = (
+    'class repr_and_html_class():\n'
+    '    def __init__(self):\n'
+    '        pass\n\n'
+    '    def __repr__(self):\n'
+    '        return "This is the __repr__"\n\n'
+    '    def _repr_html_(self):\n'
+    '        return "<div> This is the _repr_html_ div </div>"\n\n'
+    'class_inst = repr_and_html_class()\n\n'
+    'class_inst'
+)
+
+code_print_and_repr_and_html = (
+    'print("print statement")\n\n'
+    'class repr_and_html_class():\n'
+    '    def __init__(self):\n'
+    '        pass\n\n'
+    '    def __repr__(self):\n'
+    '        return "This is the __repr__"\n\n'
+    '    def _repr_html_(self):\n'
+    '        return "<div> This is the _repr_html_ div </div>"\n\n'
+    'class_inst = repr_and_html_class()\n\n'
+    'class_inst'
+)
+
+html_out = (
+    '.. only:: builder_html\n\n'
+    '    .. raw:: html\n\n'
+    '        <div> This is the _repr_html_ div </div>'
+)
+
+text_above_html = (
+    'Out:\n\n'
+    ' .. code-block:: none\n\n'
+    '    print statement\n\n\n'
+)
+
+
+def _clean_output(output):
+    is_text = '.. rst-class:: sphx-glr-script-out' in output
+
+    is_html = '.. only:: builder_html' in output
+
+    if output.isspace():
+        return ''
+    elif is_text and is_html:
+        output_test_string = "\n".join(output.strip().split("\n")[2:])
+        return output_test_string.strip()
+    elif is_text:
+        output_test_string = "\n".join(
+        [line[4:] for line in output.strip().split("\n")[6:]])
+        return output_test_string.strip()
+    elif is_html:
+        output_test_string = "\n".join(output.strip().split("\n")[-5:])
+        return output_test_string.strip()
+    
+
+@pytest.mark.parametrize('print_eval_repr, code, expected_out', [
+    pytest.param(
+        tuple(),
+        'a=2\nb=3',
+        '',
+        id='assign,()'
```

Comment:
I would not bother with `id`, then these can all easily fit on a single line like:
```
    [tuple(), 'a=2\nb=3', '']
```
Having the more compact representation will be more readable

Reply:
Yes they do all fit on one line, the only problem is that some of the input code, which use variables assigned elsewhere, are really long - and this makes interpreting the pytest output difficult.

Reply:
Ahh. Well in that case I'd still try to make it as compact as possible by not putting each var in a new line.